src/locked_queue.cpp

```diff
@@ -0,0 +1,59 @@
+#include "locked_queue.h"
+
+// Default constructor
+template<typename T>
+LockedPriorityQueue<T>::LockedPriorityQueue() {}
+
+// Max heap comparator
+template<typename T>
+bool MaxHeapComparator<T>::operator()(const T& a, const T& b) const {
+    return a > b;
+}
+
+// Min heap comparator
+template<typename T>
+bool MinHeapComparator<T>::operator()(const T& a, const T& b) const {
+    return a < b;
+}
+
+// Constructor with custom comparator
+template<typename T>
+MinHeap<T>::MinHeap() {
+    // Assign heap_ to a priority queue with a custom comparator
+    heap_ = std::priority_queue<T, std::vector<T>, MinHeapComparator<T>> {};
```

Comment:
@Ajay-26 This is where I override the default comparator for the `std::priority_queue`